plugins/utils/rest.nix

```diff
@@ -434,6 +434,7 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin {
       }
     ];
 
+    # TODO: Clarify how to handle this, add to the plugin's luaConfig or to telescope's
     extraConfigLua = lib.mkIf cfg.enableTelescope ''require("telescope").load_extension("rest")'';
```

Comment:
This is something that recently I've been wondering how we can start ordering the generated entries with dag or something. I'd like to be able to make sure certain plugins are setup after others. I don't know how we can guarantee that telescope is available when it's added to rests config or that rest is available if we add it to telescopes config.

Reply:
Is it enough to just assign a mkOverride priority ordering system for the luaConfigs?

Reply:
The best solution in a vacuum would be to use some kind of DAG to order the plugins, but implementing it could be a bit complicated (TBH I would not know where to start)

Reply:
Yeah, definitely warrants its own Issue tracking and is out of scope on this PR... just made me think about it with this particular note. I think it's fine to keep it in rest, for now.